packages/dev/sharedUiComponents/src/lines/optionsLineComponent.tsx

```diff
@@ -80,44 +88,117 @@ export class OptionsLineComponent extends React.Component<IOptionsLineComponentP
     }
 
     updateValue(valueString: string) {
-        const value = this.props.valuesAreStrings ? valueString : parseInt(valueString);
+        let value = this.props.valuesAreStrings ? valueString : parseInt(valueString);
+
+        if (isNaN(Number(value))) {
+            for (let i = 0; i < this.props.options.length; i++) {
+                if (this.props.options.at(i)?.label === valueString) {
+                    value = Number(this.props.options.at(i)?.value);
+                }
+            }
+        }
+
+        if (value === 0 && this.props.fromFontDropdown) {
+            this.setState({ addCustom: true });
+        }
+
         this._localChange = true;
 
         const store = this.props.extractValue ? this.props.extractValue(this.props.target) : this.props.target[this.props.propertyName];
 
         if (!this.props.noDirectUpdate) {
             this.props.target[this.props.propertyName] = this._remapValueOut(value as number);
         }
-        this.setState({ value: value });
 
-        if (this.props.onSelect) {
-            this.props.onSelect(value);
+        //selecting a regular option from font dropdown
+        if (value != 0 && this.props.fromFontDropdown) {
+            this.setState({ value: value });
+            if (this.props.onSelect) {
+                this.props.onSelect(value);
+            }

```

Comment:
Looks like this code is being called in any case. Can we extract it to after the if/else if/else loops? you can also do something like this:

`this.prop.onSelect && this.prop.onSelect(value)`

Reply:
This.props.onSelect(value) is not called in the case that this.props.fromFontDropdown is true (this.state.value and value are different). I tried to combine the if and the else here since that code is called in both those cases, but when I do that the dropdown stops functioning properly (I am unable to click on any of the options)